src/tools/allocator-recipe-resolver.ts

```diff
@@ -112,17 +120,19 @@ export class AllocatorRecipeResolver {
     // Iterate over all handles in the recipe set that are shared across
     // multiple recipes, and compute their restricted type and apply them to
     // all relevant handles and connections.
-    for (const value of Object.values(handleById)) {
+    for (const handleId of Object.keys(handleById)) {
       // tslint:disable-next-line: no-any
-      const {createHandle, handles} = value as any;
+      const {createHandle, store, handles} = handleById[handleId] as any;
```

Comment:
`as {Handle, AbstractStore, Handle[]}`?

Reply:
or, are you trying to turn store into a {type, direction} pair? If so you can probably type accordingly. It's good to use the types though, they pick up bugs.